In [2]:
! pip install tweepy

     ---------------------------------------- 0.0/151.7 kB ? eta -:--:--
     ---------------------------------------- 0.0/151.7 kB ? eta -:--:--
     -- ------------------------------------- 10.2/151.7 kB ? eta -:--:--
     -- ------------------------------------- 10.2/151.7 kB ? eta -:--:--
     -- ------------------------------------- 10.2/151.7 kB ? eta -:--:--
     -- ------------------------------------- 10.2/151.7 kB ? eta -:--:--
     -- ------------------------------------- 10.2/151.7 kB ? eta -:--:--
     -- ------------------------------------- 10.2/151.7 kB ? eta -:--:--
     -- ------------------------------------- 10.2/151.7 kB ? eta -:--:--
     ------- ------------------------------ 30.7/151.7 kB 62.2 kB/s eta 0:00:02
     ------- ------------------------------ 30.7/151.7 kB 62.2 kB/s eta 0:00:02
     ---------- --------------------------- 41.0/151.7 kB 67.7 kB/s eta 0:00:02
     ---------- --------------------------- 41.0/151.7 kB 67.7 kB/s eta 0:00:02
     -----------

In [3]:
import tweepy
from datetime import datetime

In [10]:

# Set your Twitter API credentials
consumer_key = 'lrDIbijtmTQ19SZZO0iErDL5m'
consumer_secret = 'eRxyT3zeBuIRM288wHnysoW5GY53PSKVCGfJFmXsEdtHC0sf3C'
access_token = '1568252321204666373-tFqkrLuMnN6q5vViPhNF65BN52wwRE'
access_token_secret = 'P0wnSeGG0f1F5ssqkmeUglcgAUVA7U3cXrhSV7mdegnun'

# Authenticate with the Twitter API
auth = tweepy.OAuthHandler(consumer_key, consumer_secret)
auth.set_access_token(access_token, access_token_secret)
api = tweepy.API(auth, wait_on_rate_limit=True)

# Specify the Twitter username (handle) and date range
twitter_username = 'CBCPolitics'
since_date = '2021-09-02'
until_date = '2021-09-19'

# Get the user's timeline (tweets) within the specified date range using v2 API
"""tweets = api.user_timeline(
    screen_name=twitter_username,
    tweet_mode='extended',
    start_time=since_date,
    end_time=until_date,
    max_results=100
)"""

tweets = api.user_timeline(screen_name=twitter_username, count=10, tweet_mode='extended')


Forbidden: 403 Forbidden
453 - You currently have access to a subset of Twitter API v2 endpoints and limited v1.1 endpoints (e.g. media post, oauth) only. If you need access to this endpoint, you may need a different access level. You can learn more here: https://developer.twitter.com/en/portal/product

In [ ]:
# Print the text of each tweet
for tweet in tweets:
    print(tweet.full_text)

In [11]:
import requests
from bs4 import BeautifulSoup


In [12]:

def scrape_tweets(username, num_tweets=10):
    url = f'https://twitter.com/{username}'

    # Send a request to the Twitter page
    response = requests.get(url)

    if response.status_code == 200:
        soup = BeautifulSoup(response.text, 'html.parser')

        # Extract tweet content
        tweets = soup.find_all('div', class_='tweet')
        for i, tweet in enumerate(tweets[:num_tweets]):
            tweet_text = tweet.find('div', class_='tweet-text').get_text(strip=True)
            print(f'Tweet {i + 1}: {tweet_text}')
            print('-' * 50)
    else:
        print(f"Failed to retrieve tweets. Status code: {response.status_code}")

# Specify the username and number of tweets to extract
username = 'CBCPolitics'
num_tweets = 10

# Scrape tweets
scrape_tweets(username, num_tweets)


Failed to retrieve tweets. Status code: 400


In [13]:
pip install twitterscraper

  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
   ---------------------------------------- 0.0/86.7 kB ? eta -:--:--
   ---------------------------------------- 0.0/86.7 kB ? eta -:--:--
   ---- ----------------------------------- 10.2/86.7 kB ? eta -:--:--
   ---- ----------------------------------- 10.2/86.7 kB ? eta -:--:--
   ---- ----------------------------------- 10.2/86.7 kB ? eta -:--:--
   -------------- ------------------------- 30.7/86.7 kB 109.5 kB/s eta 0:00:01
   -------------- ------------------------- 30.7/86.7 kB 109.5 kB/s eta 0:00:01
   -------------- ------------------------- 30.7/86.7 kB 109.5 kB/s eta 0:00:01
   ------------------ --------------------- 41.0/86.7 kB 98.5 kB/s eta 0:00:01
   ------------------ --------------------- 41.0/86.7 kB 98.5 kB/s eta 0:00:01
   ------------------ --------------------- 41.0/86.7 kB 98.5 kB/s eta 0:00:01
   ---------------------------- ----------- 61.4/86.7 kB 113.0 kB/s 

In [14]:
from twitterscraper import query_tweets

# Specify the keyword and the number of tweets to retrieve
keyword = 'Canadian federal election'
num_tweets = 10

# Query tweets
tweets = query_tweets(keyword, limit=num_tweets)

# Print tweet content
for i, tweet in enumerate(tweets):
    print(f'Tweet {i + 1}: {tweet.text}')
    print('-' * 50)


AttributeError: 'NoneType' object has no attribute 'find_all'

In [15]:
pip install GetOldTweets3


Note: you may need to restart the kernel to use updated packages.


In [16]:
import GetOldTweets3 as got

# Specify the keyword and the number of tweets to retrieve
keyword = 'Canadian federal election'
num_tweets = 10

# Query tweets
tweetCriteria = got.manager.TweetCriteria().setQuerySearch(keyword).setTopTweets(True).setMaxTweets(num_tweets)
tweets = got.manager.TweetManager.getTweets(tweetCriteria)

# Print tweet content
for i, tweet in enumerate(tweets):
    print(f'Tweet {i + 1}: {tweet.text}')
    print('-' * 50)


An error occured during an HTTP request: HTTP Error 403: Forbidden
Try to open in browser: https://twitter.com/search?q=Canadian%20federal%20election&src=typd


AttributeError: 'tuple' object has no attribute 'tb_frame'

In [19]:
pip install snscrape

   ---------------------------------------- 0.0/74.8 kB ? eta -:--:--
   ---------------------------------------- 0.0/74.8 kB ? eta -:--:--
   ----- ---------------------------------- 10.2/74.8 kB ? eta -:--:--
   ----- ---------------------------------- 10.2/74.8 kB ? eta -:--:--
   ----- ---------------------------------- 10.2/74.8 kB ? eta -:--:--
   ----- ---------------------------------- 10.2/74.8 kB ? eta -:--:--
   ----- ---------------------------------- 10.2/74.8 kB ? eta -:--:--
   ----- ---------------------------------- 10.2/74.8 kB ? eta -:--:--
   ----- ---------------------------------- 10.2/74.8 kB ? eta -:--:--
   ---------------- ----------------------- 30.7/74.8 kB 59.5 kB/s eta 0:00:01
   ---------------- ----------------------- 30.7/74.8 kB 59.5 kB/s eta 0:00:01
   --------------------- ------------------ 41.0/74.8 kB 65.5 kB/s eta 0:00:01
   --------------------- ------------------ 41.0/74.8 kB 65.5 kB/s eta 0:00:01
   -------------------------------- ------- 61.

In [21]:
import snscrape.modules.twitter as sntwitter
import pandas as pd
import time

# Created a list to append all tweet attributes(data)
attributes_container = []

# Using TwitterSearchScraper to scrape data and append tweets to list
for i,tweet in enumerate(sntwitter.TwitterSearchScraper('from:CBCNews').get_items()):
    if i>1:
        break
    attributes_container.append([tweet.date, tweet.likeCount, tweet.sourceLabel, tweet.content])
    time.sleep(1)

    
# Creating a dataframe from the tweets list above 
tweets_df = pd.DataFrame(attributes_container, columns=["Date Created", "Number of Likes", "Source of Tweet", "Tweets"])

Error retrieving https://twitter.com/i/api/graphql/7jT5GT59P8IFjgxwqnEdQw/SearchTimeline?variables=%7B%22rawQuery%22%3A%22from%3ACBCNews%22%2C%22count%22%3A20%2C%22product%22%3A%22Latest%22%2C%22withDownvotePerspective%22%3Afalse%2C%22withReactionsMetadata%22%3Afalse%2C%22withReactionsPerspective%22%3Afalse%7D&features=%7B%22rweb_lists_timeline_redesign_enabled%22%3Afalse%2C%22blue_business_profile_image_shape_enabled%22%3Afalse%2C%22responsive_web_graphql_exclude_directive_enabled%22%3Atrue%2C%22verified_phone_label_enabled%22%3Afalse%2C%22creator_subscriptions_tweet_preview_api_enabled%22%3Afalse%2C%22responsive_web_graphql_timeline_navigation_enabled%22%3Atrue%2C%22responsive_web_graphql_skip_user_profile_image_extensions_enabled%22%3Afalse%2C%22tweetypie_unmention_optimization_enabled%22%3Atrue%2C%22vibe_api_enabled%22%3Atrue%2C%22responsive_web_edit_tweet_api_enabled%22%3Atrue%2C%22graphql_is_translatable_rweb_tweet_is_translatable_enabled%22%3Atrue%2C%22view_counts_everywhere_api

ScraperException: 4 requests to https://twitter.com/i/api/graphql/7jT5GT59P8IFjgxwqnEdQw/SearchTimeline?variables=%7B%22rawQuery%22%3A%22from%3ACBCNews%22%2C%22count%22%3A20%2C%22product%22%3A%22Latest%22%2C%22withDownvotePerspective%22%3Afalse%2C%22withReactionsMetadata%22%3Afalse%2C%22withReactionsPerspective%22%3Afalse%7D&features=%7B%22rweb_lists_timeline_redesign_enabled%22%3Afalse%2C%22blue_business_profile_image_shape_enabled%22%3Afalse%2C%22responsive_web_graphql_exclude_directive_enabled%22%3Atrue%2C%22verified_phone_label_enabled%22%3Afalse%2C%22creator_subscriptions_tweet_preview_api_enabled%22%3Afalse%2C%22responsive_web_graphql_timeline_navigation_enabled%22%3Atrue%2C%22responsive_web_graphql_skip_user_profile_image_extensions_enabled%22%3Afalse%2C%22tweetypie_unmention_optimization_enabled%22%3Atrue%2C%22vibe_api_enabled%22%3Atrue%2C%22responsive_web_edit_tweet_api_enabled%22%3Atrue%2C%22graphql_is_translatable_rweb_tweet_is_translatable_enabled%22%3Atrue%2C%22view_counts_everywhere_api_enabled%22%3Atrue%2C%22longform_notetweets_consumption_enabled%22%3Atrue%2C%22tweet_awards_web_tipping_enabled%22%3Afalse%2C%22freedom_of_speech_not_reach_fetch_enabled%22%3Afalse%2C%22standardized_nudges_misinfo%22%3Atrue%2C%22tweet_with_visibility_results_prefer_gql_limited_actions_policy_enabled%22%3Afalse%2C%22interactive_text_enabled%22%3Atrue%2C%22responsive_web_text_conversations_enabled%22%3Afalse%2C%22longform_notetweets_rich_text_read_enabled%22%3Afalse%2C%22longform_notetweets_inline_media_enabled%22%3Afalse%2C%22responsive_web_enhance_cards_enabled%22%3Afalse%2C%22responsive_web_twitter_blue_verified_badge_is_enabled%22%3Atrue%7D failed, giving up.

In [24]:
import requests
from bs4 import BeautifulSoup
urls = ['https://www.cbc.ca/news/politics/carbon-tax-rebate-rebrand-1.7115300', 'https://www.cbc.ca/news/politics/elections-canada-electofacts-disinformaton-1.7081902',
       'https://www.cbc.ca/news/canada/hamilton/headlines/in-hamilton-trudeau-blames-conservatives-for-u-s-steel-mess-1.3269954']

for url in urls: 
    response = requests.get(url)
    #print(response.text)
    soup = BeautifulSoup(response.text, 'html.parser')
    title = soup.title.text
    description = soup.find('meta', {'name': 'description'})['content']
    print(f'Title: {title}')
    print(f'Description: {description}')

Title: Liberals rebranding carbon tax rebate to ensure Canadians know where the money comes from | CBC News
Description: The Liberal government is rebranding its carbon tax rebate in an effort to make it clear to Canadians where the funds are coming from.
Title: Elections Canada launches online disinformation tool to prepare voters for next federal election | CBC News
Description: Elections Canada is preparing to insulate Canadian voters from false narratives and information during the next federal election by launching an online tool to help voters cut through misinformation and disinformation about the electoral process in Canada. 
Title: In Hamilton, Trudeau blames Conservatives for U.S. Steel mess | CBC News
Description: Justin Trudeau criticized the Conservative government on Wednesday for decisions around foreign investment that lead to bleak situations like the one U.S. Steel pensioners in Hamilton find themselves in.



The provided Python code utilizes the requests library to make HTTP requests to three distinct URLs, each representing news articles on the CBC website. The BeautifulSoup library is then employed to parse the HTML content of the retrieved web pages. For each URL, the code extracts and prints the title and description of the respective news article. The title is obtained from the title tag of the HTML, while the description is retrieved from the meta tag with the attribute name 'description.' The output consists of three sets of title and description pairs, providing a concise summary of the content of each news article, covering topics such as the rebranding of a carbon tax rebate, the launch of an online disinformation tool by Elections Canada, and Justin Trudeau's criticism of the Conservative government's decisions affecting U.S. Steel pensioners in Hamilton.

The attempts with twitterscraper and GetOldTweets3 faced various issues. The twitterscraper code encountered an AttributeError related to proxy handling, which could be caused by changes in the website's structure or restrictions. Meanwhile, the GetOldTweets3 code encountered an HTTP 403 Forbidden error, indicating that the server understood the request but refused to authorize it. The code also triggered a SystemExit when facing difficulties, providing a link for manual inspection before exiting. These issues highlight the challenges of web scraping and accessing data from external sources without using official APIs. It's crucial to consider the terms of service, potential access restrictions, and the possibility of changes in the website's structure when attempting to extract information from online platforms. If the primary goal is to obtain Twitter data, it's recommended to use the official Twitter API within the bounds of its usage policies.

In addition to the challenges faced with twitterscraper and GetOldTweets3, it's important to note that utilizing the Twitter API for accessing tweet data would require a Twitter Developer account and involve associated costs of 100 USD.